src/app/controllers/users.py

```diff
@@ -1,24 +1,28 @@
 from flask import Blueprint, jsonify, request
 
-from src.app.models.user import User, teste_users_schema
+from src.app.models.user import User, users_roles_share_schema
 
 user = Blueprint('user', __name__, url_prefix='/user')
 
 @user.route("/", methods = ['GET'])
 def list_user():
     
-    list_users = User.query.all()
+    list_users = User.query.paginate(per_page=20, error_out=True)
 
-    list_users_dict = teste_users_schema.dump(list_users)
+    list_users_dict = users_roles_share_schema.dump(list_users.items)
 
-    # if data == {}:
-    #     return jsonify(list_users_dict), 200
+    return jsonify(list_users_dict), 200
 
-    # if data['name'] == "":
-        
-    #     return jsonify(list_users_dict), 200
+@user.route("/<string:name>", methods = ['GET'])
+def list_user_by_name(name):
 
+    name_cap = name.capitalize()
```

Comment:
A Busca por nomes simples está funcional, porém não consigo realizar buscas em nomes completos/compostos. 
Penso que encontrei a solução. Acho que a busca não funciona porque o name.capitalize() torna apenas a primeira letra maiúscula e o resto minúsculo, fazendo Luiz Gustavo virar Luiz gustavo, causando o erro na hora da busca. Recomendo colocar: 
```name_cap = name.title()```
Em um teste rápido aqui no local, parece ter resolvido o problema. Aguardo feedback de vocês.

Reply:
Surgiu um segundo problema, que nomes compostos que contenham palavras com letra minúscula, como "João da Silva" não sejam buscados.

Reply:
Vai ter que dar um split no nome, criar um array das exceptions: "da" "de"  percorrer um por um no for e fazer o if para ignorar os exceptions, elif para arrumar se o cara escrever "Da" ou "De" e else para deixar a primeira letra maíuscula(.title)

Reply:
Obrigado pelas dicas, com base nisso eu com ajuda de Vinicius, conseguimos refatorar todo o código da endpoint para ficar mais lógica e simples, agora pode ser pesquisado tanto com "Da" ou "da" e etc... Caso a pessoa não passe também nenhum nome/número, ele irá retornar uma paginação por padrão, ou seja, mesmo que não seja enviado nada no query param ele irá retornar uma página com pelo menos 2 usuários nela, caso ele queira avançar de página, é possível passando o número da página que deseja.